File: backend/src/services/ai/evidence.js
/**
 * src/services/ai/evidence.js
 *
 * ─── Purpose ───────────────────────────────────────────────────────────────
 * Integration service for structured evidence collection AI queries.
 * ───────────────────────────────────────────────────────────────────────────
 */

import aiClient from './client.js';

/**
 * Triggers structured evidence collection for a resolved company entity.
 *
 * Returns the full response envelope containing both data and meta blocks.
 *
 * @param {object} resolverResult - Resolved company profile data from Phase 5
 * @returns {Promise<{ status: string, data: object, meta: object }>}
 */
export const collectEvidence = async (resolverResult) => {
  const response = await aiClient._request('/api/v1/evidence/collect', {
    method: 'POST',
    body: JSON.stringify(resolverResult),
  });
  return response;
};

File: ai-service/src/api/routes/evidence.py
# ─────────────────────────────────────────────────────────────────
# src/api/routes/evidence.py
# ─────────────────────────────────────────────────────────────────
#
# Route definitions for structured evidence collection.
# Maps POST /collect to the coordinator service pipeline.
# ─────────────────────────────────────────────────────────────────

from fastapi import APIRouter
from src.services.company_service import ResolverResult
from src.services.evidence_service import collect_evidence_package
from src.utils.response import send_success, send_error
from src.constants.error_codes import AI_SERVICE_TIMEOUT

router = APIRouter()

@router.post("/collect")
def collect_evidence(payload: ResolverResult):
    """
    POST /api/v1/evidence/collect
    
    Accepts a resolved company entity (ResolverResult) and triggers the
    multi-channel structured evidence collection pipeline.
    """
    try:
        evidence_package = collect_evidence_package(payload.model_dump())
        return send_success(data=evidence_package.model_dump())
    except TimeoutError as te:
        return send_error(
            code=AI_SERVICE_TIMEOUT,
            message=str(te),
            status_code=504
        )
